bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java

```diff
@@ -0,0 +1,836 @@
+            updateState(CHANNEL_DOWN_LIGHT_BRIGHTNESS, new DecimalType(updateState.downLightBrightness));
+
+            updateState(CHANNEL_FLAME, new DecimalType(updateState.flame));
+            updateState(CHANNEL_FP_FAN_STATE, updateState.fpfanPower == 0 ? OnOffType.OFF : OnOffType.ON);
+            updateState(CHANNEL_FP_FAN_SPEED, new DecimalType(updateState.fpfanSpeed));
+
+            updateState(CHANNEL_OPEN_CLOSE, updateState.open == 0 ? OpenClosedType.CLOSED : OpenClosedType.OPEN);
+
+            // Mark all the stateless channels stopped
+            updateState(CHANNEL_LIGHT_START_STOP, OnOffType.OFF);
+            updateState(CHANNEL_LIGHT_DIRECTIONAL_INC, OnOffType.OFF);
+            updateState(CHANNEL_LIGHT_DIRECTIONAL_DECR, OnOffType.OFF);
+            updateState(CHANNEL_UP_LIGHT_START_STOP, OnOffType.OFF);
+            updateState(CHANNEL_UP_LIGHT_DIRECTIONAL_INC, OnOffType.OFF);
+            updateState(CHANNEL_UP_LIGHT_DIRECTIONAL_DECR, OnOffType.OFF);
+            updateState(CHANNEL_DOWN_LIGHT_START_STOP, OnOffType.OFF);
+            updateState(CHANNEL_DOWN_LIGHT_DIRECTIONAL_INC, OnOffType.OFF);
+            updateState(CHANNEL_DOWN_LIGHT_DIRECTIONAL_DECR, OnOffType.OFF);
+            updateState(CHANNEL_STOP, OnOffType.OFF);
+            updateState(CHANNEL_HOLD, OnOffType.OFF);
+
+        } else {
+            logger.debug("No state information provided to update channels with");
+        }
+    }
+
+    private boolean hasConfigurationError() {
+        ThingStatusInfo statusInfo = getThing().getStatusInfo();
+        return statusInfo.getStatus() == ThingStatus.OFFLINE
+                && statusInfo.getStatusDetail() == ThingStatusDetail.CONFIGURATION_ERROR;
+    }
+
+    private synchronized boolean wasBindingUpdated() {
+        // Check if the binding has been updated
+        boolean updatedBinding = true;
+        @Nullable
+        String lastBindingVersion = this.getThing().getProperties().get(PROPERTIES_BINDING_VERSION);
+        updatedBinding = !CURRENT_BINDING_VERSION.equals(lastBindingVersion);
+        if (updatedBinding) {
+            logger.info("Bond Home binding has been updated.");
+            logger.info("Current version is {}, prior version was {}.", CURRENT_BINDING_VERSION, lastBindingVersion);
+
+            // Update the thing with the new property value
+            final Map<String, String> newProperties = new HashMap<>(thing.getProperties());
+            newProperties.put(PROPERTIES_BINDING_VERSION, CURRENT_BINDING_VERSION);
+
+            final ThingBuilder thingBuilder = editThing();
+            thingBuilder.withProperties(newProperties);
+            updateThing(thingBuilder.build());
+        }
+        return updatedBinding;
+    }
+
+    private synchronized boolean wasThingUpdatedExternally(BondDevice devInfo) {
+        // Check if the Bond hash tree has changed
+        config = getConfigAs(BondDeviceConfiguration.class);
```

Comment:
When the config changes `initialize` is called again and that already sets the `config`. Therefor it's not needed to set `config` again here? Or did I miss some behavior here?